ligo/haskell/src/Ligo/Typescript.hs

```diff
@@ -0,0 +1,440 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Ligo.Typescript (generateTs) where
+
+import Universum
+
+import Data.Char (toUpper)
+import qualified Data.Text as T
+import Fmt (Buildable(build), blockListF', (+|), (|+))
+import qualified Fmt
+import System.FilePath
+
+import Lorentz.Entrypoints
+import Michelson.Typed.Annotation
+import Michelson.Typed.Entrypoints (pnNotes)
+import Michelson.Untyped.Annotation
+
+-- | This is mostly same as the 'Notes t' type, but without the type
+-- parameter and without annotations that we are not interested.
+data EpType
+  = EpString
+  | EpInt
+  | EpNat
+  | EpUnit
+  | EpSignature
+  | EpOperation
+  | EpContract EpType
+  | EpLambda EpType EpType
+  | EpMap EpType EpType
+  | EpBigMap EpType EpType
+  | EpBytes
+  | EpAddress
+  | EpKey
+  | EpChainId
+  | EpMutez
+  | EpBool
+  | EpSet EpType
+  | EpKeyHash
+  | EpTimestamp
+  | EpList EpType
+  | EpOption EpType
+  | EpPair AnnotatedField AnnotatedField
+  | EpOr AnnotatedField AnnotatedField
+  deriving stock (Show, Generic)
+
+type AnnotatedField
+ = (FieldAnn, EpType)
```

Comment:
I have mixed feelings about the usage of types like this because GHC often trips up and shows error messages like `Expected (FieldAnn, EpType) but got whatever` instead of using `AnnotatedField`. Apart from this, I think a `data` looks better than a `type` aliasing a tuple.

Reply:
Yes, changed.